src/components/utilityFunctions.ts

```diff
@@ -87,15 +87,16 @@ export const filterObject = (
   return Object.fromEntries(filtered)
 }
 
+// Generic function for recursively crawling the file system. Will run method "fileOperation" on all files within "directory"
 export const crawlFileSystem = async (
-  newPath: string,
-  fileOperation: (newPath: string) => void
+  directory: string,
+  fileOperation: (filePath: string) => void
```

Comment:
Better names -- you were using "newPath" for two different things, which is a bit confusing.